dot/parachain/dispute/participation.go

```diff
@@ -0,0 +1,376 @@
+package dispute
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	parachain "github.com/ChainSafe/gossamer/dot/parachain/runtime"
+
+	"github.com/ChainSafe/gossamer/dot/parachain/dispute/overseer"
+	"github.com/ChainSafe/gossamer/dot/parachain/dispute/types"
+	parachainTypes "github.com/ChainSafe/gossamer/dot/parachain/types"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+// CandidateComparator comparator for ordering of disputes for candidate.
+type CandidateComparator struct {
+	relayParentBlockNumber *uint32
+	candidateHash          common.Hash
+}
+
+// NewCandidateComparator creates a new CandidateComparator.
+func NewCandidateComparator(relayParentBlockNumber *uint32,
+	receipt parachainTypes.CandidateReceipt,
+) (CandidateComparator, error) {
+
+	candidateHash, err := receipt.Hash()
+	if err != nil {
+		return CandidateComparator{}, fmt.Errorf("hash candidate receipt: %w", err)
```

Comment:
i don't see you using `CandidateComparator` instance in case of error. you can return nil.

Reply:
The reason is because I don't want to return a pointer here. It's interesting, could you tell me why you'd prefer it the other way in this case?